services/brig/test/integration/API/User/Account.hs

```diff
@@ -1632,18 +1633,91 @@ testTooManyMembersForLegalhold opts brig = do
         const 403 === statusCode
         const (Right "too-many-members-for-legalhold") === fmap Wai.label . responseJsonEither
 
+testDeleteUserWithCompletelyDeletedUser :: Brig -> Cannon -> AWS.Env -> Http ()
+testDeleteUserWithCompletelyDeletedUser brig cannon aws = do
+  u <- randomUser brig
+  liftIO $ Util.assertUserJournalQueue "user activate testDeleteUserWithCompletelyDeletedUser" aws (userActivateJournaled u)
+  setHandleAndDeleteUser brig cannon u [] aws $
+    \uid -> deleteUserInternal uid brig !!! const 202 === statusCode
+  do
+    let uid = userId u
+    deleteUserInternal uid brig
+      !!! do
+        const 200 === statusCode
+
+testDeleteUserWithNoUser :: Brig -> Http ()
+testDeleteUserWithNoUser brig = do
+  nonExistingUid :: UserId <- liftIO $ generate arbitrary
+  deleteUserInternal nonExistingUid brig
+    !!! do
+      const 404 === statusCode
+
+testDeleteUserWithNotDeletedUser :: HasCallStack => Brig -> Cannon -> AWS.Env -> Http ()
+testDeleteUserWithNotDeletedUser brig cannon aws = do
+  u <- randomUser brig
+  liftIO $ Util.assertUserJournalQueue "user activate testDeleteUserWithNotDeletedUser" aws (userActivateJournaled u)
+  do
+    setHandleAndDeleteUser brig cannon u [] aws $
+      ( \uid' ->
+          deleteUserInternal uid' brig
+            !!! do
+              const 202 === statusCode
+      )
+
+testDeleteUserWithDanglingProperty :: Brig -> Cannon -> AWS.Env -> Http ()
+testDeleteUserWithDanglingProperty brig cannon aws = do
+  u <- randomUser brig
+  liftIO $ Util.assertUserJournalQueue "user activate testDeleteUserWithDanglingProperty" aws (userActivateJournaled u)
+
+  let uid = userId u
+  -- First set a unique handle (to verify freeing of the handle)
+  hdl <- randomHandle
+  let update = RequestBodyLBS . encode $ HandleUpdate hdl
+  put (brig . path "/self/handle" . contentJson . zUser uid . zConn "c" . body update)
+    !!! const 200 === statusCode
+
+  deleteUserInternal uid brig !!! const 202 === statusCode
+  liftIO $ Util.assertUserJournalQueue "user deletion testDeleteUserWithDanglingProperty" aws (userDeleteJournaled uid)
+
+  setProperty brig (userId u) "foo" objectProp
+    !!! const 200 === statusCode
+  getProperty brig (userId u) "foo" !!! do
+    const 200 === statusCode
+    const (Just objectProp) === responseJsonMaybe
+
+  execAndAssertUserDeletion brig cannon u (Handle hdl) [] aws $ \uid' -> do
+    deleteUserInternal uid' brig
+      !!! do
+        const 202 === statusCode
+
+  getProperty brig (userId u) "foo" !!! do
+    const 404 === statusCode
+  where
+    objectProp =
+      object
+        [ "key.1" .= ("val1" :: Text),
+          "key.2" .= ("val2" :: Text)
+        ]
+
 -- helpers
 
 setHandleAndDeleteUser :: Brig -> Cannon -> User -> [UserId] -> AWS.Env -> (UserId -> HttpT IO ()) -> Http ()
 setHandleAndDeleteUser brig cannon u others aws execDelete = do
```

Comment:
What an odd set of things to group into one function...?  But ok!

Reply:
Well, it's sole purpose is to reduce code duplication... :shrug:
Nice side-effect is, that some assumptions are checked for (almost) all deletion tests, that likely wouldn't be if one had to duplicate the assertions.

(Though, the `And` in the function name is almost always an indicator that concepts are mixed that shouldn't be... :thinking: )